srml/support/src/dispatch.rs

```diff
@@ -67,37 +67,62 @@ pub trait Parameter: Codec + Clone + Eq {}
 #[cfg(not(feature = "std"))]
 impl<T> Parameter for T where T: Codec + Clone + Eq {}
 
-/// Declare a module struct and implement the dispatch logic.
+/// Declares a `Module` struct and a `Call` enum, which implements the dispatch logic.
 ///
-/// Usually used as follows:
+/// ## Example
 ///
+/// ```rust,ignore
 /// decl_module! {
-///		pub struct Module<T: Trait> for enum Call where origin: T::Origin
-///{}
-///	}
+/// 	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+/// 		fn foobar(origin, ...) -> Result {
+///				...
+/// 			OK(())
+/// 		}
+///			pub fn foo(origin) -> Result {...}  // public function easily accessible to other modules
+///			fn bar(origin) {...}				// macro expands to return `Result`
+///			fn baz() -> Result {...}			// privileged function
+///		}
+/// }
+/// ```
+///
+/// *Where `...` indicates truncated code for example brevity.*
+///
+/// Function declaration is set with the header where:
+///
+/// * `Module`: The struct generated by the macro, with type `Trait`.
+/// * `Call`: The enum generated for every module, which implements [`Callable`](./dispatch/trait.Callable.html).
+/// * `origin`: Alias of `T::Origin`, declared by the [`impl_outer_origin!`](./macro.impl_outer_origin.html) macro.
+/// * `Result`: The expected return type from module functions.
+///
+/// Function declaration variations include:
+///
+/// * [Optional] Public functions: `pub` functions are easily accessible to other modules, e.g. `foo`.
```

Comment:
Can we just name the functions more readable names rather than foo, bar, baz?

Like `my_pub_function`, `my_privileged_function`, etc...